crates/bevy_reflect/bevy_reflect_derive/src/derive_data.rs

```diff
@@ -0,0 +1,191 @@
+use crate::container_attributes::ReflectAttrs;
+use crate::field_attributes::{
+    parse_field_attrs, ReflectFieldAttr, REFLECT_ATTRIBUTE_NAME, REFLECT_VALUE_ATTRIBUTE_NAME,
+};
+use crate::utility::get_bevy_reflect_path;
+use syn::{Data, DataStruct, DeriveInput, Field, Fields, Generics, Ident, Meta, Path};
+
+pub enum DeriveType {
+    Struct,
+    TupleStruct,
+    UnitStruct,
+    Value,
+}
+
+/// Data used by derive macros for `Reflect` and `FromReflect`
+///
+/// # Example
+/// ```ignore
+/// //                          attrs
+/// //        |----------------------------------------|
+/// #[reflect(PartialEq, Serialize, Deserialize, Default)]
+/// //            type_name       generics
+/// //     |-------------------||----------|
+/// struct ThingThatImReflecting<T1, T2, T3> {
+///     x: T1, // |
+///     y: T2, // |- fields
+///     z: T3  // |
+/// }
+/// ```
+pub struct ReflectDeriveData<'a> {
+    derive_type: DeriveType,
+    attrs: ReflectAttrs,
+    type_name: &'a Ident,
+    generics: &'a Generics,
+    fields: Vec<(&'a Field, ReflectFieldAttr, usize)>,
```

Comment:
Yeah you're probably right. I don't know why I didn't do this to begin with...

Reply:
I'll keep the `ReflectFieldAttr`, though, instead of the `is_active` field. The reason for this is because the original PR that this is based on extends the `ReflectFieldAttr` struct with a new attribute.